apps/dashboard/src/components/primitives/field-editor/variable-popover/hooks/use-variable-parser.ts

```diff
@@ -0,0 +1,72 @@
+import { useMemo } from 'react';
+import { TransformerWithParam } from '../types';
+import { TRANSFORMERS } from '../constants';
+
+export function useVariableParser(variable: string) {
```

Comment:
Can't we use liquid library for all this parsing? 
I'm not sure if that's the right thing but seems like that https://liquidjs.com/api/classes/Tokenizer.html#constructor